plugin/pkg/proxy/connect.go

```diff
@@ -117,6 +118,20 @@ func (p *Proxy) Connect(ctx context.Context, state request.Request, opts Options
 	for {
 		ret, err = pc.c.ReadMsg()
 		if err != nil {
+			// If the error is an overflow, we probably have a misbehaving upstream
+			// that is sending out-of-spec (>512 byte) UDP responses without an eDNS0
+			// OPT RR. Set the truncate bit and return the result - this will make the
+			// client retry over TCP (if that's supported) or at least receive a clean
+			// error. The connection is still good so we break before the close.
+			if proto == "udp" && strings.Contains(err.Error(), "overflow") {
```

Comment:
So I recreated a scenario in which upstream sets the truncated flag, but doesn't truncate the response. I get a `buffer size too small` error instead of overflow. Would we want to support retrying for this scenario?

Reply:
what's the buffer size set to?